spec/bson/big_decimal_spec.rb

```diff
@@ -0,0 +1,317 @@
+# Copyright (C) 2016-2021 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "spec_helper"
+
+describe BSON::BigDecimal do
+
+  describe '#from_bson' do
+    shared_examples_for 'a BSON::BigDecimal deserializer' do
+
+      let(:decimal128) do
+        BSON::Decimal128.new(argument)
+      end
+
+      let(:from_bson) do
+        BigDecimal.from_bson(decimal128.to_bson)
+      end
+
+      let(:expected_from_bson) do
```

Comment:
Usually when we use the word "expected" in a test, it refers to a literal value (e.g.  a string that is immediately specified) rather than a calculated value. "Expected" values should generally be clear as to what they are and are easy to verify for correctness.

In these tests, the variables marked "expected" are actually calculated values and it isn't obvious what they are. I suggest using a different term for them, for example you could say that this is a "decimal128 decoded value" or something similar.

Reply:
done